app/assets/javascripts/shopify_app/redirect.js

```diff
@@ -12,15 +15,23 @@
       // If the current window is the 'parent', change the URL by setting location.href
       window.top.location.href = targetInfo.url;
     } else {
-      // If the current window is the 'child', change the parent's URL with postMessage
+      // If the current window is the 'child' or embedded, change the parent's URL with App Bridge redirect
+      // This case happens when the app is installed but the app does not know that because there's no shop entry
```

Comment:
Not sure about this. I think the main reason we need to redirect out of the iframe is when the app wants to go through oauth to get new scopes.

Reply:
The new access scopes seem to only kick off re-auth when a backend call fails, but I will test this out some more. I usually delete the shop entry in the app's db to trigger this flow.